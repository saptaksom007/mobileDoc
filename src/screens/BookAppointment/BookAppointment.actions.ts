import { OptionType } from './BookAppointment.types'

export const ActionTypes = {
  INIT_BOOKING_APPOINTMENT: 'INIT_BOOKING_APPOINTMENT',
  SELECT_BOOK_APPOINTMENT_OPTION: 'SELECT_BOOK_APPOINTMENT_OPTION',
  NAVIGATE_TO_CHAT: 'NAVIGATE_TO_CHAT',
  NAVIGATE_TO_BOOKAPPOINTMENT: 'NAVIGATE_TO_BOOKAPPOINTMENT',
  CHANGE_REASON_FOR_APPOINTMENT: 'CHANGE_REASON_FOR_APPOINTMENT',
  CHANGE_OTHER_DOCTOR: 'CHANGE_OTHER_DOCTOR',
}

export default class Actions {
  static initBookingAppointment() {
    return {
      type: ActionTypes.INIT_BOOKING_APPOINTMENT,
      payload: undefined,
    }
  }

  static selectOption(option: { [key: string]: OptionType }) {
    return {
      type: ActionTypes.SELECT_BOOK_APPOINTMENT_OPTION,
      payload: option,
    }
  }

  static navigateToChat(doctorId: string, conversationId: number) {
    return {
      type: ActionTypes.NAVIGATE_TO_CHAT,
      payload: { doctorId, conversationId },
    }
  }

  static navigateToBookAppointment(
    conversationId: number,
    clinicId: number,
    scheduleAppointment?: number,
    futureAppointmentsCount?: number,
  ) {
    return {
      type: ActionTypes.NAVIGATE_TO_BOOKAPPOINTMENT,
      payload: {
        conversationId,
        clinicId,
        scheduleAppointment,
        futureAppointmentsCount,
      },
    }
  }

  static changeReasonForAppointment(text: string) {
    return {
      type: ActionTypes.CHANGE_REASON_FOR_APPOINTMENT,
      payload: text,
    }
  }

  static changeOtherDoctor(text: string) {
    return {
      type: ActionTypes.CHANGE_OTHER_DOCTOR,
      payload: text,
    }
  }
}
