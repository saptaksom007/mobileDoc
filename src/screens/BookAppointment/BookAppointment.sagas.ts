import { put, takeLatest, select } from 'redux-saga/effects'
import { pipe, propOr, prop, pathOr } from 'ramda'
import i18n from 'ex-react-native-i18n'
import moment from 'moment'

import ConversationsListActions from 'screens/ConversationsList/ConversationsList.actions'
import { Actions as NavAction } from 'navigation/SagaNavigation'
import toString from 'common-docdok/lib/utils/toString'
import Actions, { ActionTypes } from './BookAppointment.actions'
import { UserDtoType } from 'common-docdok/lib/types'

const getConversationById = (
  { messaging: { conversations } }: any,
  conversationId: any,
) => pipe(propOr({}, conversationId))(conversations)

const getDoctorNameById = ({ userCache: { users } }: any, doctorId: any) =>
  pipe<{ [key: string]: UserDtoType }, UserDtoType, string>(
    prop(doctorId),
    (u: UserDtoType) => toString.person(u),
  )(users)

const getOption = (state: any, key: any) =>
  pipe(pathOr('', ['bookappointment', key]), k =>
    i18n.t(`bookappointment.message.${k}`),
  )(state)

const getReasonForAppointment = pathOr('', [
  'bookappointment',
  'reasonForAppointment',
])

const getOtherDoctor = pathOr('', ['bookappointment', 'otherDoctor'])

const getChatText = ({ date, day, time, doctor, reasonForAppointment }: any) =>
  i18n.t(
    `bookappointment.message.content${
      reasonForAppointment ? 'WithReason' : ''
    }`,
    {
      date,
      day,
      time,
      doctor,
      reasonForAppointment,
    },
  )

const isAnyDoctorSelected = ({ bookappointment: { preferredDoctor } }: any) =>
  preferredDoctor === 0

function* navigateToBookAppointment({
  payload: { conversationId, clinicId, scheduleAppointment },
}: any) {
  yield put(
    NavAction.push('bookappointment', {
      conversationId,
      clinicId,
      scheduleAppointment,
    }),
  )
}

function* navigateToChat({ payload: { doctorId, conversationId } }: any) {
  const conversation = yield select(getConversationById, conversationId)
  const isAnyDoctor = yield select(isAnyDoctorSelected)
  const doctorSelected = yield select(getDoctorNameById, doctorId)
  const doctor = isAnyDoctor
    ? i18n.t('bookappointment.message.anyDoctor')
    : doctorSelected
  const otherDoctor = yield select(getOtherDoctor)
  const date = yield select(getOption, 'preferredDate')
  const day = yield select(getOption, 'preferredDayOfWeek')
  const time = yield select(getOption, 'preferredTimeOfDay')
  const reasonForAppointment = yield select(getReasonForAppointment)
  yield put(
    ConversationsListActions.gotoChatRequested(
      Number(conversation.id),
      getChatText({
        doctor: doctor || otherDoctor,
        date,
        day,
        time,
        reasonForAppointment,
      }),
    ),
  )
}

function* selectOption({
  payload: { preferredDate, preferredDayOfWeek },
}: any) {
  if (preferredDate && preferredDate === 'urgent') {
    yield put(
      NavAction.showLocalWarning(i18n.t('bookappointment.urgentWarning')),
    )
  }

  const numOfDay = yield select(
    pathOr(0, ['bookappointment', 'preferredDayOfWeek']),
  )
  const preferredDateState = yield select(
    pathOr(0, ['bookappointment', 'preferredDate']),
  )
  const dayOfWeek = moment().day()
  if (
    (preferredDayOfWeek &&
      preferredDayOfWeek <= dayOfWeek &&
      preferredDateState === 'thisWeek') ||
    (preferredDate &&
      preferredDate === 'thisWeek' &&
      numOfDay > 0 &&
      numOfDay <= dayOfWeek)
  ) {
    yield put(
      Actions.selectOption({
        preferredDate: 'withinTwoWeeks',
      }),
    )
  }
}

export default [
  takeLatest(ActionTypes.NAVIGATE_TO_CHAT, navigateToChat),
  takeLatest(
    ActionTypes.NAVIGATE_TO_BOOKAPPOINTMENT,
    navigateToBookAppointment,
  ),
  takeLatest(ActionTypes.SELECT_BOOK_APPOINTMENT_OPTION, selectOption),
]
