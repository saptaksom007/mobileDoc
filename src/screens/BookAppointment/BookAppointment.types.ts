import {
  BookAppointmentDate,
  BookAppointmentDayOfWeek,
  BookAppointmentTime
} from 'api/types'

export type OptionType =
  | BookAppointmentDate
  | BookAppointmentDayOfWeek
  | BookAppointmentTime
  | number
  | undefined
  | boolean

export interface PreferredOption {
  preferredDate?: BookAppointmentDate
  preferredDayOfWeek?: BookAppointmentDayOfWeek
  preferredTimeOfDay?: BookAppointmentTime
  preferredDoctor?: number
}
