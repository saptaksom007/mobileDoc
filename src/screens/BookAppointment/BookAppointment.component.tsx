import React, { Component } from 'react'

import {
  View,
  ScrollView,
  InteractionManager,
  StyleSheet,
  Platform,
} from 'react-native'
import { connect } from 'react-redux'
import { compose } from 'recompose'
import i18n from 'ex-react-native-i18n'
import { Divider, Button, FormInput } from 'react-native-elements'

import {
  BookAppointmentDate,
  BookAppointmentDayOfWeek,
  BookAppointmentTime,
  BookAppointmentDateValues,
  BookAppointmentDayOfWeekValues,
  BookAppointmentTimeValues,
} from 'api/types'
import withSubtitle from 'screens/enhancers/withSubtitle'
import withKeyboardAvoidingView from 'screens/enhancers/withKeyboardAvoidingView'
import { Color } from 'constants/Color'
import Options from './components/Options/Options.component'
import {
  fontFamily,
  fontFamilyBold,
} from 'components/CustomText/CustomText.styles'
import { Heading3 } from 'components/CustomText/CustomText.component'

import { getTestId } from 'utilities/environment'
import ScheduleAppointmentActions from 'screens/ScheduleAppointment/ScheduleAppointment.actions'

import Actions from './BookAppointment.actions'

import { OptionType } from './BookAppointment.types'
import { Action } from 'redux'
import { NavigationScreenProp } from 'react-navigation'
import { isDayStillInThisWorkWeek } from 'utilities/date'

const styles = StyleSheet.create({
  container: {},
  preference: {
    paddingVertical: 7,
  },
  requestBtn: {
    paddingHorizontal: 34,
  },
  subtitle: {
    marginLeft: 10,
    color: Color.black,
    fontFamily,
  },
  formInputText: {
    color: '#000',
  },
  formInput: {
    marginLeft: 10,
    marginRight: 10,
    marginBottom: Platform.select({ ios: 15, android: 0 }),
    backgroundColor: '#f0efef',
    color: '#c4c4c4',
    borderRadius: 5,
  },
})

const requestAppointmentTestId = getTestId('request-appointment')
const inputRequestAppointmentTestId = getTestId('input-request-appointment')

const mapName = (doctor: any) =>
  `${
    doctor.salutation.endsWith('.')
      ? doctor.salutation
      : `${doctor.salutation}.`
  }${
    doctor.firstName ? ` ${doctor.firstName.substr(0, 1).toUpperCase()}.` : ''
  } ${doctor.lastName}`

const mapValue = (value: any) =>
  Object.values(value).map((key: any) => i18n.t(`bookappointment.${key}`))

const MAX_LENGTH = 128

interface Props {
  dispatch(action: Action): void
  navigation: NavigationScreenProp<any>
  preferredDate: BookAppointmentDate
  preferredDayOfWeek: BookAppointmentDayOfWeek
  preferredTimeOfDay: BookAppointmentTime
  preferredDoctor: number
  doctors: any[]
  doctorsOptions: string[]
  otherDoctor: string
  scheduleAppointment?: number
  other?: boolean
}

interface State {
  doctorsOptions: string[]
}

export class BookAppointment extends Component<Props, State> {
  static defaultProps: {
    doctorsOptions: []
  }

  componentDidMount() {
    const { dispatch } = this.props
    InteractionManager.runAfterInteractions(() => {
      dispatch(Actions.initBookingAppointment())
    })
  }

  otherDoctorInput: any

  selectOption = (option: { [key: string]: OptionType }) =>
    this.props.dispatch(Actions.selectOption(option))

  scrollView: any

  navigateToChat = () => {
    const {
      dispatch,
      preferredDoctor,
      doctors,
      navigation: {
        state: {
          params: { conversationId },
        },
      },
      scheduleAppointment,
    } = this.props

    dispatch(
      Actions.navigateToChat(
        doctors[preferredDoctor - 1] && doctors[preferredDoctor - 1].userRef,
        conversationId,
      ),
    )
    InteractionManager.runAfterInteractions(() => {
      if (scheduleAppointment) {
        dispatch(
          ScheduleAppointmentActions.scheduleAppointment(scheduleAppointment),
        )
      }
      dispatch(Actions.initBookingAppointment())
    })
  }

  selectOtherDoctor = (index?: number) => {
    const { doctorsOptions } = this.props
    this.selectOption({
      preferredDoctor: index,
      other: index === doctorsOptions.length - 1,
    })
    InteractionManager.runAfterInteractions(() => {
      if (this.otherDoctorInput) {
        this.otherDoctorInput.focus()
      }
    })
  }

  render() {
    const {
      preferredDate,
      preferredDayOfWeek,
      preferredTimeOfDay,
      preferredDoctor,
      doctorsOptions,
      otherDoctor,
      other,
      dispatch,
    } = this.props

    const disabledButton =
      preferredDate === 'urgent' ||
      (other &&
        ((otherDoctor && otherDoctor.trim() === '') ||
          otherDoctor === undefined ||
          otherDoctor === '')) ||
      (preferredDate === 'thisWeek' &&
        preferredDayOfWeek !== 'anyDayOfWeek' &&
        !isDayStillInThisWorkWeek(preferredDayOfWeek))

    return (
      <View style={{ flex: 1 }}>
        <ScrollView
          style={{ flex: 1 }}
          contentContainerStyle={styles.container}
          keyboardShouldPersistTaps='never'
        >
          <Options
            style={styles.preference}
            title={i18n.t('bookappointment.preferredDate')}
            values={mapValue(BookAppointmentDateValues)}
            selectedIndex={BookAppointmentDateValues.indexOf(preferredDate)}
            onPress={index =>
              this.selectOption({
                preferredDate: BookAppointmentDateValues[index],
              })
            }
          />
          <Divider />
          <Options
            style={styles.preference}
            title={i18n.t('bookappointment.preferredDayOfWeek')}
            values={mapValue(BookAppointmentDayOfWeekValues)}
            selectedIndex={BookAppointmentDayOfWeekValues.indexOf(
              preferredDayOfWeek,
            )}
            onPress={index =>
              this.selectOption({
                preferredDayOfWeek: BookAppointmentDayOfWeekValues[index],
              })
            }
          />
          <Divider />
          <Options
            style={styles.preference}
            title={i18n.t('bookappointment.preferredTimeOfDay')}
            values={mapValue(BookAppointmentTimeValues)}
            selectedIndex={BookAppointmentTimeValues.indexOf(
              preferredTimeOfDay,
            )}
            onPress={index =>
              this.selectOption({
                preferredTimeOfDay: BookAppointmentTimeValues[index],
              })
            }
          />
          <Divider />
          <Options
            style={styles.preference}
            title={i18n.t('bookappointment.preferredDoctor')}
            values={doctorsOptions}
            selectedIndex={preferredDoctor}
            onPress={index => this.selectOtherDoctor(index)}
          />
          {other && (
            <View>
              <Divider />
              <View
                style={styles.preference}
                testID={inputRequestAppointmentTestId}
                accessibilityLabel={inputRequestAppointmentTestId}
              >
                <Heading3 style={styles.subtitle}>
                  {i18n.t('bookappointment.otherDoctorTitle')}
                </Heading3>
                <FormInput
                  ref={ref => (this.otherDoctorInput = ref)}
                  maxLength={MAX_LENGTH}
                  onChangeText={text =>
                    dispatch(Actions.changeOtherDoctor(text))
                  }
                  placeholder={i18n.t('bookappointment.otherDoctorPlaceholder')}
                  underlineColorAndroid={'#f0efef'}
                  inputStyle={styles.formInputText}
                  containerStyle={styles.formInput}
                  autoCorrect={false}
                />
              </View>
            </View>
          )}
          <Divider />
          <View style={styles.preference}>
            <Heading3 style={styles.subtitle}>
              {i18n.t('bookappointment.reasonForAppointment')}
            </Heading3>
            <FormInput
              testID={inputRequestAppointmentTestId}
              accessibilityLabel={inputRequestAppointmentTestId}
              maxLength={MAX_LENGTH}
              onChangeText={text =>
                dispatch(Actions.changeReasonForAppointment(text))
              }
              placeholder={i18n.t(
                'bookappointment.reasonForAppointmentPlaceholder',
              )}
              underlineColorAndroid={'#f0efef'}
              inputStyle={styles.formInputText}
              containerStyle={styles.formInput}
              autoCorrect={false}
            />
          </View>

          <Button
            testID={requestAppointmentTestId}
            accessibilityLabel={requestAppointmentTestId}
            style={styles.requestBtn}
            title={i18n.t('bookappointment.requestAppointment')}
            icon={{ name: 'check' }}
            backgroundColor={'#8950b7'}
            disabled={disabledButton}
            fontFamily={fontFamilyBold}
            onPress={this.navigateToChat}
            buttonStyle={{
              borderRadius: 12,
            }}
          />
        </ScrollView>
      </View>
    )
  }
}

function select({
  bookappointment: {
    preferredDate,
    preferredDayOfWeek,
    preferredTimeOfDay,
    preferredDoctor,
    otherDoctor,
    clinicId,
    other,
  },
  healthrelationCache: { professionals },
}: any): any {
  const professionalsValues: any[] = Object.values(professionals)
  const doctors =
    (clinicId &&
      professionalsValues.filter((doc: any) => doc.clinicId === clinicId)) ||
    []
  const doctorsOptions = [
    i18n.t('bookappointment.anyDoctor'),
    ...doctors.map(mapName),
    i18n.t('bookappointment.otherDoctor'),
  ]
  return {
    preferredDate,
    preferredDayOfWeek,
    preferredTimeOfDay,
    preferredDoctor,
    doctors,
    doctorsOptions,
    otherDoctor,
    other,
  }
}

export default compose<Props, any>(
  connect(select),
  withKeyboardAvoidingView,
  withSubtitle('bookappointment.subtitle'),
)(BookAppointment)
