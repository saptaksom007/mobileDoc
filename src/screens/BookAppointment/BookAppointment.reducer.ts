import { ActionTypes } from './BookAppointment.actions'

const {
  INIT_BOOKING_APPOINTMENT,
  SELECT_BOOK_APPOINTMENT_OPTION,
  CHANGE_REASON_FOR_APPOINTMENT,
  CHANGE_OTHER_DOCTOR,
  NAVIGATE_TO_BOOKAPPOINTMENT
} = ActionTypes

const initialState = {
  error: undefined,
  preferredDate: 'anyDate',
  preferredDayOfWeek: 'anyDayOfWeek',
  preferredTimeOfDay: 'anyTime',
  preferredDoctor: 0,
  reasonForAppointment: undefined,
  otherDoctor: undefined,
  conversationId: undefined,
  clinicId: undefined
}

const cleanFreeTextSafe = (text: string) =>
  text && text.replace(':', ' ').replace('/n', ' ')

export default function bookappointment(
  state: typeof initialState = initialState,
  action: { type: keyof typeof ActionTypes; payload: any }
) {
  switch (action.type) {
    case INIT_BOOKING_APPOINTMENT: {
      const { conversationId, clinicId } = state
      return { ...initialState, conversationId, clinicId }
    }

    case SELECT_BOOK_APPOINTMENT_OPTION: {
      return { ...state, ...action.payload }
    }

    case CHANGE_REASON_FOR_APPOINTMENT: {
      const reasonForAppointment = cleanFreeTextSafe(action.payload)
      return { ...state, reasonForAppointment }
    }

    case CHANGE_OTHER_DOCTOR: {
      const otherDoctor = cleanFreeTextSafe(action.payload)
      return { ...state, otherDoctor }
    }

    case NAVIGATE_TO_BOOKAPPOINTMENT: {
      const { conversationId, clinicId } = action.payload
      return { ...state, conversationId, clinicId }
    }

    default:
      return state
  }
}
