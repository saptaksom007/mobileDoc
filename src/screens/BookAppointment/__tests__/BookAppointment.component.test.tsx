import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { BookAppointment } from '../BookAppointment.component'
import { NavigationScreenProp } from 'react-navigation'

it('<BookAppointment/> render correctly', () => {
  const tree = renderer
    .create(
      <BookAppointment
        dispatch={() => {}}
        doctors={[]}
        preferredDate="thisMonth"
        preferredDayOfWeek="friday"
        preferredTimeOfDay="afternoon"
        preferredDoctor={0}
        otherDoctor=""
        doctorsOptions={[]}
        navigation={
          { state: { params: { conversationId: 1 } } } as NavigationScreenProp<
            any,
            any
          >
        }
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
