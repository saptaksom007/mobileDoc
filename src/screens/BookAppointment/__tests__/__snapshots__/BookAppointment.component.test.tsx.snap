// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookAppointment/> render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={Object {}}
    keyboardShouldPersistTaps="never"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "paddingVertical": 7,
          }
        }
      >
        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontFamily": "TitilliumWeb-SemiBold",
                "fontSize": 18.125,
                "lineHeight": 22,
              },
              Object {
                "color": "#363c40",
                "fontFamily": "TitilliumWeb-Regular",
                "marginLeft": 10,
              },
            ]
          }
        >
          bookappointment.preferredDate
        </Text>
        none
      </View>
      <View
        style={
          Object {
            "paddingVertical": 7,
          }
        }
      >
        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontFamily": "TitilliumWeb-SemiBold",
                "fontSize": 18.125,
                "lineHeight": 22,
              },
              Object {
                "color": "#363c40",
                "fontFamily": "TitilliumWeb-Regular",
                "marginLeft": 10,
              },
            ]
          }
        >
          bookappointment.preferredDayOfWeek
        </Text>
        none
      </View>
      <View
        style={
          Object {
            "paddingVertical": 7,
          }
        }
      >
        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontFamily": "TitilliumWeb-SemiBold",
                "fontSize": 18.125,
                "lineHeight": 22,
              },
              Object {
                "color": "#363c40",
                "fontFamily": "TitilliumWeb-Regular",
                "marginLeft": 10,
              },
            ]
          }
        >
          bookappointment.preferredTimeOfDay
        </Text>
        none
      </View>
      <View
        style={
          Object {
            "paddingVertical": 7,
          }
        }
      >
        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontFamily": "TitilliumWeb-SemiBold",
                "fontSize": 18.125,
                "lineHeight": 22,
              },
              Object {
                "color": "#363c40",
                "fontFamily": "TitilliumWeb-Regular",
                "marginLeft": 10,
              },
            ]
          }
        >
          bookappointment.preferredDoctor
        </Text>
        none
      </View>
      <View
        style={
          Object {
            "paddingVertical": 7,
          }
        }
      >
        <Text
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 17.5,
                "lineHeight": 22,
                "overflow": "visible",
              },
              Object {
                "color": "rgba(27, 106, 158, 0.85)",
                "fontFamily": "TitilliumWeb-SemiBold",
                "fontSize": 18.125,
                "lineHeight": 22,
              },
              Object {
                "color": "#363c40",
                "fontFamily": "TitilliumWeb-Regular",
                "marginLeft": 10,
              },
            ]
          }
        >
          bookappointment.reasonForAppointment
        </Text>
      </View>
      none
    </View>
  </RCTScrollView>
</View>
`;
